source/beamline/BioXAS/BioXASSideBeamStatusControl.cpp

```diff
@@ -0,0 +1,185 @@
+#include "BioXASSideBeamStatusControl.h"
+#include "actions3/AMActionSupport.h"
+
+BioXASSideBeamStatusControl::BioXASSideBeamStatusControl(const QString &name, QObject *parent) :
+	BioXASBeamStatusControl(name, parent)
+{
+	// Initialize inherited variables.
+
+	setContextKnownDescription("SideBeamStatus");
+
+	// Initialize local variables.
+
+	frontEndStatus_ = 0;
+	poeBeamStatus_ = 0;
+	soeBeamStatus_ = 0;
+
+	// Current settings.
+
+	updateStates();
+}
+
+BioXASSideBeamStatusControl::~BioXASSideBeamStatusControl()
+{
+
+}
+
+bool BioXASSideBeamStatusControl::shouldStop() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->shouldStop();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::canMeasure() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->canMeasure();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::canMove() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->canMove();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::canStop() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->canStop();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::isOn() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->isOn();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::isOff() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->isOff();
+
+	return result;
+}
+
+void BioXASSideBeamStatusControl::setFrontEndBeamStatus(BioXASFrontEndBeamStatusControl *newControl)
+{
+	if (frontEndStatus_ != newControl) {
+
+		if (poeBeamStatus_)
+			poeBeamStatus_->setFrontEndBeamStatus(0);
+
+		frontEndStatus_ = newControl;
+
+		if (poeBeamStatus_)
+			poeBeamStatus_->setFrontEndBeamStatus(frontEndStatus_);
+
+		updateStates();
+
+		emit frontEndBeamStatusChanged(frontEndStatus_);
+	}
+}
+
+void BioXASSideBeamStatusControl::setPOEBeamStatus(BioXASSidePOEBeamStatusControl *newControl)
+{
+	if (poeBeamStatus_ != newControl) {
+
+		if (poeBeamStatus_)
+			disconnect( poeBeamStatus_, 0, this, 0 );
```

Comment:
Same thing here. The SOE depends on the POE and should be updated with the new POE control set.